profiling: add documentation for the config file and CLI arguments (#4398)
elastic/observability-docs: docs/en/observability/profiling-advanced-configuration.asciidoc

* <<profiling-no-kernel-version-check, Override kernel version check >>: Configure the Universal Profiling Agent to bypass the kernel version compatibility check.
* <<profiling-envs, Environment variables for the Universal Profiling Agent >>: Configure the Universal Profiling Agent using the environment.

> WARNING: Command line arguments to the Universal Profiling Agent have precedence over environment variables. And environment varaibles have precedence over configuration files.

Reviewer: ```suggestion
WARNING: Command line arguments to the Universal Profiling Agent have precedence over environment variables. And environment variables have precedence over the configuration file.

```

| `-tags` | `string` | `-tags='cloud_region:us-central1;env:staging'` |  Set specific tags. See <<profiling-tag-data-query, Tag data for querying>> for more details.
| `-project-id` | `uint` | `-project-id 73` | Splits profiling data into logical groups that you control. You can assign any non-zero, unsigned integer <= 4095.
| `-secret-token` | `string` | `-secret-token=abc123` | Set the secret token for the communication with the Universal Profiling Collector to `abc123`.
| `-collection-agent` | `string` | `-collection-agent=example.com:443` |  Set the destination for reporting profiling information to `example.com:443`.
| `-probabilistic-interval` | `duration` | `-probabilistic-interval=2m30s`| Set the probabilistic interval to `2m30s`. See <<profiling-probabilistic-profiling, Probabilistic profiling>> for more details.
| `-probabilistic-threshold` | `uint` | `-probabilistic-threshold=50` | Set the probabilistic threshold to `50`. See <<profiling-probabilistic-profiling, Probabilistic profiling>> for more details.

> | `-config` | `string` | `-config=/opt/Elastic/universal-profiling/pf-host-agent.conf` | Set the path for the configuration file of the Universal Profiling Agent. See <<profiling-config-file, Configuration file of the Universal Profiling Agent>> for more details.

Reviewer: ```suggestion
| `-config` | `string` | `-config=/etc/Elastic/universal-profiling/pf-host-agent.conf` | Set the path for the configuration file of the Universal Profiling Agent. See <<profiling-config-file, Configuration file of the Universal Profiling Agent>> for more details.

```

* <<profiling-no-kernel-version-check, Override kernel version check >>: Configure the Universal Profiling Agent to bypass the kernel version compatibility check.
* <<profiling-envs, Environment variables for the Universal Profiling Agent >>: Configure the Universal Profiling Agent using the environment.

> WARNING: Command line arguments to the Universal Profiling Agent have precedence over environment variables. And environment variables have precedence over the configuration file.

Reviewer 2: ```suggestion
WARNING: Command line arguments to the Universal Profiling Agent take precedence over environment variables, and environment variables take precedence over the configuration file.

```

The Universal Profiling Agent accepts the following CLI arguments:

[options="header"]
|==================================
| CLI argument | Type | Example | Description
| `-v` | `bool` | `-v` | Run the Universal Profiling Agent in verbose mode.
| `-no-kernel-version-check` | `bool` | `-no-kernel-version-check` | Disable the kernel version check. See <<profiling-no-kernel-version-check, Override kernel version check >> for more details.
| `-tags` | `string` | `-tags='cloud_region:us-central1;env:staging'` |  Set specific tags. See <<profiling-tag-data-query, Tag data for querying>> for more details.
| `-project-id` | `uint` | `-project-id 73` | Splits profiling data into logical groups that you control. You can assign any non-zero, unsigned integer <= 4095.

> | `-secret-token` | `string` | `-secret-token=abc123` | Set the secret token for the communication with the Universal Profiling Collector to `abc123`.

Reviewer 2: ```suggestion
| `-secret-token` | `string` | `-secret-token=abc123` | Set the secret token for communicating with the Universal Profiling Collector to `abc123`.

```
